plenum/server/monitor.py

```diff
@@ -118,13 +121,19 @@ def add_instance(self, inst_id):
     def remove_instance(self, instId):
         for req in self._requests.values():
             req.remove_instance(instId)
-        reqs_to_del = [key for key, req in self._requests.items() if req.is_ordered_by_all]
-        for req in reqs_to_del:
-            del self._requests[req]
-            self._unordered.discard(req)
-            self._handled_unordered.discard(req)
+        keys_to_del = [key for key, req in self._requests.items() if req.is_ordered_by_all]
+        for key in keys_to_del:
+            self.force_req_drop(key)
         self.instances_ids.remove(instId)
 
+    def force_req_drop(self, key):
+        if key in self._requests:
+            del self._requests[key]
+        if key in self._unordered:
+            self._unordered.discard(key)
+        if key in self._handled_unordered:
+            self._handled_unordered.discard(key)
```

Comment:
When discarding some value from set there is no need to check that it was here in the first place

Reply:
I believe that checking that `force_req_drop` doesn't break anything inside `RequestTracker` is done by `test_req_drop_*` group of tests that call this function in a various stages of request processing and then check that pool is functional.

Reply:
Added tests.